mountpoint-s3/src/prefetch.rs

```diff
@@ -703,6 +751,23 @@ mod tests {
             let read_size = (size as usize / read_factor).max(1);
             run_sequential_read_test(default_stream(), size, read_size, config);
         }
+
+        #[test]
+        fn proptest_sequential_read_with_cache(
+            size in 1u64..1 * 1024 * 1024,
+            read_size in 1usize..1 * 1024 * 1024,
+            block_size in 16usize..1 * 1024 * 1024,
+            config: TestConfig,
+        ) {
+            run_sequential_read_test(caching_stream(block_size), size, read_size, config);
+        }
+
+        #[test]
+        fn proptest_sequential_read_small_read_size_with_cache(size in 1u64..1 * 1024 * 1024, read_factor in 1usize..10,
+            block_size in 16usize..1 * 1024 * 1024, config: TestConfig) {
+            let read_size = (size as usize / read_factor).max(1);
+            run_sequential_read_test(caching_stream(block_size), size, read_size, config);
+        }
```

Comment:
We have both because it's annoying to write "choose a random read size _up to_ the object size" with the nice proptest syntax. It's just laziness on my part when I wrote the tests these new versions are copied and pasted from. Let's fix it separately.

Reply:
Ah, got it! I'm happy with a comment that points the reader in that direction.